### PR TITLE
A quick and dirty fix of CTAS failures: unable to rename from viewfs

### PR DESCRIPTION
**Pull request by beinan**
Seeing the failure below when run CTAS on viewfs

`Unable to rename from viewfs://hadoop-dw2-nn.smf1.twitter.com/user/beinanw/warehouse/aaa/.hive-staging/538ee740-6cce-46b7-96a7-ca0fbfc52148 to viewfs://hadoop-dw2-nn.smf1.twitter.com/user/beinanw/warehouse/aaa: target directory already exists`

It seems that presto creates the temporary folder - ".hive-staging/{id}" just under the target folder, which would also create the target folder in advance if the target folder did not exist.  But for the new tables (such as the table crated by CTAS), the target folder should not be created until the prepareAddTable():SemiTransactionalHiveMetastore got called, otherwise we will see the failure above.

To fix this failure,  I make it check the existence of the target folder first.  If the target folder is not there, the temporary folder is created as {target_folder}/../.hive-staging/{id} rather than {target_folder}/.hive-staging/{id}.   In this case, the temporary folder will be created in the sibling folder of the target folder.  So the target folder won't be created in advance.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by beinan** [comment]
Sorry for the delay of this one,  test case added.  @zhenxiao

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java [hunk 1]
```diff
@@ -13,18 +13,25 @@
...
 import static com.facebook.presto.hive.HiveWriteUtils.isViewFileSystem;
-import static com.facebook.presto.testing.TestingConnectorSession.SESSION;
+import static org.testng.Assert.assertEquals;
 import static org.testng.Assert.assertFalse;
 import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.fail;
 
 public class TestHiveWriteUtils
 {
+    //HdfsContext CONTEXT = new HdfsContext(new ConnectorIdentity("test", Optional.empty(), Optional.empty()));
```
Comment (zhenxiao) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java:34]: do we need this line?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java [hunk 2]
```diff
@@ -48,4 +55,39 @@ public void testIsViewFileSystem()
...
+    {
+        HdfsEnvironment hdfsEnvironment = createTestHdfsEnvironment(new HiveClientConfig(), new MetastoreClientConfig());
+        Path viewfsPath = new Path("viewfs://ns-default/test-dir");
+
+        // ViewFS check requires the mount point config
+        hdfsEnvironment.getConfiguration(CONTEXT, viewfsPath).set("fs.viewfs.mounttable.ns-default.link./test-dir", "file://" + Files.createTempDir());
+
+        Path temporaryPath = createTemporaryPath(SESSION, CONTEXT, hdfsEnvironment, viewfsPath);
+
+        assertEquals(temporaryPath.getParent().toString(), "viewfs://ns-default/test-dir/.hive-staging");
```
Comment (zhenxiao) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java:69]: could we have testcases for both targetPath exist and not exist?

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java (hunk 1 above)
Comment (beinan) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java]: removed

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java (hunk 2 above)
Comment (beinan) [reply presto-hive/src/test/java/com/facebook/presto/hive/TestHiveWriteUtils.java:71]: test cases added

**Pull request merged**